mlflow/entities/span.py

```diff
@@ -391,6 +392,77 @@ def to_immutable_span(self) -> "Span":
         # All state of the live span is already persisted in the OpenTelemetry span object.
         return Span(self._span)
 
+    @classmethod
+    def from_immutable_span(
```

Comment:
note: This is where the copying happens. We need to do the copy by actually calling `start_detached_span()` API instead of directly instantiation spans, because OpenTelemetry hard blocks calling Span constructor manually.